### PR TITLE
En_Fr Frogs 47 of 48 func decomped

### PR DESCRIPTION
**Pull request by engineer124**
New PR as I accidentally did some git shenanigans with my other PR.

### CODE REVIEW

**Comment by fig02** [comment]
refer to this closed pr for previous review comments https://github.com/zeldaret/oot/pull/511

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 1]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B22C.s")
+void EnFr_ChildSong(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frog;
+    u8 songIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B2C8.s")
+    if (this->jumpCounter < 48) {
+        if (this->jumpCounter % 4 == 0) {
+            // 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3 (11 celebratory jumps)
+            EnFr_SetupJumpingUp(this, sJumpOrder[(this->jumpCounter >> 2) & 7]); // % 8
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:738]: comment probably not needed, someone can look at the data if they want to know the implementation details

Comment (fig02) [reply src/overlays/actors/ovl_En_Fr/z_en_fr.c:738]: and if the mod works you should use it. if it doesnt match then you can leave it as & and prob remove the comment

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 2]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
+        }
+    } else {
+        songIndex = this->songIndex;
+        if (songIndex == FROG_STORMS) {
+            this->actor.textId = 0x40AA;
+            EnFr_SetupReward(this, globalCtx, false);
+        } else if ((gSaveContext.eventChkInf[13] & sSongIndex[songIndex]) == 0) {
+            frog = sEnFrPointers.frogs[sSongToFrog[songIndex]];
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:746]: you can use ! for the flag check

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 3]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
+        frog = sEnFrPointers.frogs[frogIndex];
+        frog->isButterflyDrawn = false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BB34.s")
+// Frogs 2 song is either predetermined (HP not obtained) or random (HP obtained)
+u8 EnFr_GetNextNoteFrogSong(u8 ocarinaNoteIndex) {
+    if ((gSaveContext.eventChkInf[13] & 1) == 0) {
+        return gFrogsSongPtr[ocarinaNoteIndex];
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:820]: ! here as well

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 4]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BCB4.s")
+s32 EnFr_IsFrogSongComplete(EnFr* this, GlobalContext* globalCtx) {
+    u8 index;
+    u8 ocarinaNote;
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    u8 ocarinaNoteIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BD94.s")
+    if (this->ocarinaNote == (*msgCtx).unk_E410) { // Using (*msgCtx.unk_E410) was the only way to get it to work???
+        this->ocarinaNoteIndex++;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:846]: the message context temp is prob not real either, but we can revisit this at a later time

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 5]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BDFC.s")
+void EnFr_OcarinaMistake(EnFr* this, GlobalContext* globalCtx) {
+    func_80106CCC(globalCtx);
+    this->reward = GI_NONE;
+    func_80078884(NA_SE_SY_OCARINA_ERROR);
+    func_800ED858(0); // Comment from z_en_fu.c: seems to be related to setting instrument type (Disable ocarina)
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:867]: you can probably omit this comment, we'll have a better idea of what it does with audio decomp

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 6]
```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C1E4.s")
+    func_800ED858(0); // Comment from z_en_fu.c: seems to be related to setting instrument type (Disable ocarina)
+    globalCtx->msgCtx.msgMode = 0x37;
+    this->actionFunc = EnFr_PrintTextBox;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C248.s")
+void EnFr_PrintTextBox(EnFr* this, GlobalContext* globalCtx) {
+    func_8010B680(globalCtx, this->actor.textId, &this->actor); // Looking up text box message
+    this->actionFunc = EnFr_TalkBeforeReward;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:939]: can omit comment. not sure what looking up means, i think its actually displaying it. but yea we can wait for message decomp

**Review by fig02 — COMMENTED** [review]
nice, just a few things then this is good to go from me.
you might also need to merge master since there are some bgcheck things in here

**Review by fig02 — APPROVED** [review]
updated and reviewed

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 7]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+
+void EnFr_OrientUnderwater(EnFr* this) {
+    Vec3f vec1;
+    Vec3f vec2;
+    f32 xzDist;
+
+    vec1.x = vec1.y = 0.0f;
+    xzDist = sLogSpotToFromWater[this->actor.params].xzDist;
+    this->xzDistToLogSpot = xzDist;
+    vec1.z = xzDist;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:217]:
```suggestion
    vec1.z = this->xzDistToLogSpot = sLogSpotToFromWater[this->actor.params].xzDist;
```

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 8]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+            break;
+        case 10:
+            this->songIndex = FROG_STORMS;
+            break;
+        case 4:
+            EnFr_OcarinaMistake(this, globalCtx);
+            break;
+        case 1:                                   // Ocarina note played, but no song played
+            switch (globalCtx->msgCtx.unk_E410) { // Jumping frogs in open ocarina based on ocarina note played
+                case FROG_BTN_A:
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:706]: This enum seems wrong to have in this actor if it's for a message context field

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 9]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+            this->reward = GI_RUPEE_PURPLE;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C474.s")
+void EnFr_Deactivate(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frogLoop1;
+    EnFr* frogLoop2;
+    int frogIndex;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:980]:
```suggestion
    s32 frogIndex;
```

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 10]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C524.s")
+    // Originally was going to have separate butterfly actor
+    // Changed to include butterfly as part of frog actor
+    // This unused code would have frozen the butterfly actor above frog
+    if (this->unusedButterflyActor) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:985]:
```suggestion
    if (this->unusedButterflyActor != NULL) {
```

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 11]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+    if (!(gSaveContext.eventChkInf[13] & 1)) {
+        return gFrogsSongPtr[ocarinaNoteIndex];
+    } else {
+        return sOcarinaNotes[(s32)Rand_ZeroFloat(60.0f) % 5];
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BC14.s")
+void EnFr_SetupFrogSong(EnFr* this, GlobalContext* globalCtx) {
+    if (this->frogSongTimer) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:823]:
```suggestion
    if (this->frogSongTimer != 0) {
```

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 12]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+        this->skelAnime.playSpeed = 1.0f;
+    }
+}
+
+void EnFr_OrientOnLogSpot(EnFr* this, GlobalContext* globalCtx) {
+    s16 rotYRemaining = Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 10000, 100);
+
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+
+    if (!rotYRemaining && this->skelAnime.curFrame == this->skelAnime.endFrame) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:458]:
```suggestion
    if ((rotYRemaining == 0) && (this->skelAnime.curFrame == this->skelAnime.endFrame)) {
```

File: src/overlays/actors/ovl_En_Fr/z_en_fr.c [hunk 13]
```diff
@@ -27,99 +138,963 @@ const ActorInit En_Fr_InitVars = {
...
+void EnFr_DecrementBlinkTimer(EnFr* this) {
+    if (this->blinkTimer) {
+        this->blinkTimer--;
+    } else {
+        this->blinkFunc = EnFr_DecrementBlinkTimerUpdate;
+    }
+}
+
+void EnFr_DecrementBlinkTimerUpdate(EnFr* this) {
+    if (this->blinkTimer) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Fr/z_en_fr.c:399]:
```suggestion
    if (this->blinkTimer != 0) {
```

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**